src/lib.rs

```diff
@@ -118,6 +124,40 @@ impl OpenatDirExt for openat::Dir {
             }
         }
     }
+
+    fn ensure_dir_all<P: openat::AsPath>(&self, p: P, mode: libc::mode_t) -> io::Result<()> {
+        let p = p
+            .to_path()
+            .ok_or_else(|| io::Error::new(io::ErrorKind::InvalidInput, "nul byte in file name"))?;
+        let p = p.as_ref();
+        // Convert to a `Path` basically just so that we can call `.parent()` on it.
+        let p = Path::new(OsStr::from_bytes(p.to_bytes()));
+        // Our first system call here is a direct `mkdirat()` - if that succeeds or
+        // we get EEXIST then we're done.
+        match self.create_dir(p, mode) {
+            Ok(_) => return Ok(()),
+            Err(e) => match e.kind() {
+                io::ErrorKind::AlreadyExists => {}
```

Comment:
Inconsistent Ok flow between the `return Ok(())` above and the fallthrough to the `Ok(())` below.

Reply:
Thanks, fixed!